src/services/user.ts

```diff
@@ -79,8 +83,9 @@ export class UserService implements IUserService {
       throw new BadRequestError('Email already taken');
     }
     const credits = NEW_USER_CREDITS;
-    const salt = await bcrypt.genSalt(5);
-    const passwordHash = await bcrypt.hash(password, salt);
+    const hashedPassword = await this.generateSaltAndHash(password);
```

Comment:
`const { salt, hash: passwordHash } = await this.generateSaltAndHash(password);`